### ISSUE TITLE
Relative positions are potentially broken after drifting steps

### ISSUE THREAD

**Issue by yszhang95** [yszhang95/tred#12]
The relative positions in time domain are potentially broken after drifting steps.

In the [line](https://github.com/brettviren/tred/blob/c1e60a5b5df0ef5f8975bacdd6960ef64c5ff3d4/src/tred/graph.py#L92), tail is passed in and transported to the target (anode plane).

However, the meaning of tail is obscure. It can means:
- X0 (X1), the start (end) point. 
  - In this case, X0 may be closer or further to anode than X1. Let us assume there are two steps, at the same initial position. One from X0 to X1 points to anode and the other one points to cathode. Then, the `Drifter` in `tred.graph` gives the different drift time. The former one has longer drift time and is collected while the later one has a shorter time.
- The point closer/further to anode among X0 or X1.
  - There is no ambiguity in this case. The case discussed above does not appear. However, the input is generally not ordered by their distance to anode, but the direction of particle trajectory. Users need to swap the orders by themselves, like `X0X1 = torch.where(X0[:,0] > X0[:,1], torch.hstack([X0,X1]), torch.hstack([X1,X0]))` and usage of `>` or `<` depends on the drift direction. This additional steps should be warned to user, for example, in the docstring.
  - This definition of using point closer to anode, is also consistent with the next operation, convolution. Signal should start convolution when the first element arrives at the response plane. It gives a smaller diffusion width but the inaccuracy should be small.

The action items:
- [x] Implement docstring to reminder users what is the definition of `tail` and `head`.
- ~Converting `X0` and `X1` to `tail` and `head` in `TPCDataset`, for example, `collate_fn`.~

**Comment by yszhang95** [comment]
The meaning of `tail` and `head` in Drifter are interchangeable. The meaning of output `dtail` and `dhead` are defined to be
- `dtail`, The post-drift position of the initial point closer to the anode.
- `dhead`, The post-drift position of the initial point farther from the anode.

Related commits are
- 7a762673f104d054d6f53697d86c021165431f37
- 00677cbbd63a92e200319010fed708c719bfdb6b (finalized version)

**Comment by yszhang95** [comment]
Merged in #14.

**Comment by yszhang95** [comment]
Another solution is to use mid-point between X0 and X1, which is also plausible. The class `Raster` then needs updated accordingly.

The advantage of using the point closer to anode is that time is precise for convolution with response. However, diffusion expand the region of interests and bring the inaccuracy again. And as the step length become much smaller than diffusion, the difference between X0/X1 and midpoint are unimportant and these two approaches are equivalent.
